pages/scripts/exporthardcopy.js

```diff
@@ -0,0 +1,271 @@
+                    '<tr class="lh_jhc_form"><td>&nbsp;</td></tr>'+
+                    '<tr class="lh_jhc_form"><td>&nbsp;</td></tr>'+
+                    '<tr class="lh_jhc_form"><td>&nbsp;</td></tr>'+
+                    '<tr class="lh_jhc_form"><td>Any Further Work Needed:</td></tr>'+
+                    '<tr class="lh_jhc_form"><td>&nbsp;</td></tr>'+
+                    '<tr class="lh_jhc_form"><td>&nbsp;</td></tr>'+
+                      /*
+                    '<tr>'+
+                      '<td>'+
+                        '<table class="jobHardcopy_tasker">'+
+                          '<tr><td>++tasker++<br/>Email: ++tasker_email++<br/>Phone: ++tasker_phone++<br/>Fax: ++tasker_fax++</td></tr>'+
+                          '<tr><td></td></tr>'+
+                        '</table>'+
+                      '</td>'+
+                    '</tr>'+
+                      */
+                  '</table>'+
+                  '<hr/>';
+
+//make the call to beacon
+function HackTheMatrix(id, host, progressBar) {
+  console.log('HackTheMatrix( %s , %s , %o )',id, host, progressBar);
+  var unit = [];
+  var start = new Date(decodeURIComponent(params.start));
+  var end = new Date(decodeURIComponent(params.end));
+
+  if(selectedcolumns.length == 0){
+    console.log('No Fields Selected');
+    return false;
+  }else{
+    console.log('selectedcolumns = %o' , selectedcolumns);
+  }
+
+  if (typeof id !== 'undefined') {
+    id.split(',').forEach(function(d){
+      var newObj = {'Id': d};
+      unit.push(newObj);
+    });
+  }
+
+  if(unit.length == 0){
+    console.log('No Units Selected');
+  }else{
+    console.log(unit);
+  }
+
+
+  LighthouseJob.get_json(unit, host, start, end, function(jobs) {
+
+    $.each(jobs.Results , function(k,v){
+      var thisJob = jobTemplate;
+      _.each(selectedcolumns, function(key){
+        // Raw Value
+        var rawValue = key.split('.').reduce(function(obj,i) {return obj[i]}, v);
+        // Special Cases
+        switch(key){
+          case 'JobReceived':
+            var rawDate = new Date(v.JobReceived);
+            rawDate = new Date(rawDate.getTime() + ( rawDate.getTimezoneOffset() * 60000 ));
+            rawValue = (''+rawDate).split( ' GMT' )[0];
+            break;
+          case 'Tags':
+            rawValue = v.Tags.map(function(v){return v.Name}).join(' , ');
+            break;
+          case 'JobPriorityType.Name':
+            rawValue = ( rawValue == 'General' ? '' : ' ('+rawValue+')' );
+            break;
+          case 'PermissionToEnterPremises':
+            rawValue = ( rawValue ? 'P' : '<strong>NO</strong> P' )+'ermission granted to access premises if resident is absent';
+            break;
+        }
+        if( rawValue == null )
+          rawValue = '';
+        thisJob = thisJob.replace( '++' + key + '++' , rawValue );
+      });
+      thisJob = thisJob.replace( /\+\+[^\+]+\+\+/g , '<em>Unknown or Unavailable</em>' );
```

Comment:
It'd be

> /^\+\+[^+]+\+\+$/g

for the whole string wouldn't it? (With the "$" to the left of the RegExp delimiter.)

Not looking to match the whole string, just placeholders within the template. But I take onboard that I don't need to escape within the character class.